core/src/main/java/org/elasticsearch/index/query/RegexpQueryParser.java

```diff
@@ -76,7 +76,7 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                         currentFieldName = parser.currentName();
                     } else {
                         if ("value".equals(currentFieldName)) {
-                            value = parser.objectBytes();
+                            value = parser.textOrNull();
```

Comment:
Alex's comment on the prefix query that we need to check for null anyway and can have a better error message this way makes sense to me